1. What is the northernmost airport in the United States?


In [1]:
import pandas as pd
import csv
import matplotlib.pyplot as plt
import urllib.request
import matplotlib 
import io
import requests


%matplotlib inline


In [8]:
import numpy as np
import pandas as pd
nor_airport = pd.read_csv('airports.csv')
nor_airport.head()
nor_airport.sort_values("lat", ascending = False).head(5)

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

Based on the data "Dilliant Hopkins Airpot" is the northernmost airport in the United States.But as we can see the longitude is 42.89 and latitude is 72.27, there seems to be an error in data entry.I checked the data of it in google and data provided there is different.
Hence,**"Wiley Post Will Rogers Mem"** is considered as the northernmost airport in the U.S based on the given report.


*SOURCE:* 
 - https://latitude.to/articles-by-country/us/united-states/179120/dillanthopkins-airport

2.What is the easternmost airport in the United States?

In [13]:
east= pd.read_csv('airports.csv')
east_airport = east[(east['lat']> 44) & (east['lat'] <= 50) & (east['lon'] >= -75) & (east['lon'] <= -65)]
east_airport.head()
east_airport.sort_values('lon', ascending=False).head(5)


faa                                 name        lat        lon  alt  tz  \
444   EPM           Eastport Municipal Airport  44.910111 -67.012694   45  -5   
624   HUL                         Houlton Intl  46.123083 -67.792056  489  -5   
259   CAR                         Caribou Muni  46.871500 -68.017917  626  -5   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958 -68.044797  534  -5   
1398  WFK  Northern Aroostook Regional Airport  47.285556 -68.312778  988  -5   

     dst             tzone  
444    A  America/New_York  
624    A  America/New_York  
259    A  America/New_York  
1101   A  America/New_York  
1398   A  America/New_York

 **Eastport Municipal Airport** is the earthernmost airport in the United States. I double checked the longitude and latitude in multiple websites and it is correct as mentioned above.
 
*SOURCE:* 
- https://www.lat-long.com/
- https://latitude.to/

3. On February 12th, 2013, which New York area airport had the windiest weather?


In [16]:
windy = pd.read_csv('weather.csv')
# sorting data based on requirements
windy_weather = windy[(windy["month"]==2) & (windy["day"]==12) & (windy["year"]==2013)]
windy_weather.head()
windy_weather.sort_values("wind_speed", ascending=False).head(6)

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   
1008     EWR  2013      2   12     2  39.92  28.04  62.21     270.0   

       wind_speed  wind_gust  precip  pressure  visib             time_hour  
1009   1048.36058        NaN     0.0    1008.3   10.0  2013-02-12T08:00:00Z  
18417    23.01560   31.07106     0.0    1007.1   10.0  2013-02-12T07:00:00Z  
1018     21.86482   31.07106     0.0    1012.5   10.0  2013-02-12T17:00:00Z  
18428    21.86482   25.31716     0.0    1011.7   10.0  2013-02-12T18:00:00Z  
18429    20.71404   25.31716     0.0    1011.5   10.0  2013-02-12T19:00:00Z  
1008     20.71404   25.31716     0.0    1007.8   10.0  2013-02-12T07:00:00Z

Based on the report,**LGA** had the windiest weather.